vue/sbc-common-components/src/services/token.services.ts

```diff
@@ -92,26 +103,36 @@ class TokenServices {
         })
         .error(() => {
           clearTimeout(this.timerId)
-          console.log('refresh failed')
         })
     }, refreshInMilliSeconds)
-    console.log('[TokenServices] Timer id:' + this.timerId)
   }
 
   private syncSessionStorage () {
-    sessionStorage.setItem('KEYCLOAK_TOKEN', this.kc.token)
-    sessionStorage.setItem('KEYCLOAK_REFRESH_TOKEN', this.kc.refreshToken)
-    sessionStorage.setItem('KEYCLOAK_ID_TOKEN', this.kc.idToken)
+    if (this.kc) {
+      if (this.kc.token != null) {
+        sessionStorage.setItem('KEYCLOAK_TOKEN', this.kc.token)
+      }
+      if (this.kc.refreshToken != null) {
+        sessionStorage.setItem('KEYCLOAK_REFRESH_TOKEN', this.kc.refreshToken)
+      }
+      if (this.kc.idToken != null) {
+        sessionStorage.setItem('KEYCLOAK_ID_TOKEN', this.kc.idToken)
+      }
+    }
   }
 
   decodeToken () {
     try {
       let token = sessionStorage.getItem('KEYCLOAK_TOKEN')
-      const base64Url = token.split('.')[1]
-      const base64 = decodeURIComponent(window.atob(base64Url).split('').map(function (c) {
-        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)
-      }).join(''))
-      return JSON.parse(base64)
+      if (token != null) {
+        const base64Url = token.split('.')[1]
```

Comment:
This will throw a runtime exception if the token is invalid.  Maybe an additional check here.

Reply:
totally unused function.. I might as well just remove it